currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -97,6 +98,10 @@ class ResCurrencyRateProvider(models.Model):
         )
     ]
 
+    @api.model
+    def _get_close_date(self):
+        return False
```

Comment:
Hmm I'm very sorry but this is the best way to have further problems.

Look, if you return a "time" here... it's OK in some time zone, but not OK in others. Also, if you split the date and time fields, you'll find yourself in weird corner cases, like somebody in Australia wanting to update currencies from BCE once a day, but it will never get data because when he starts the cron, it will be tomorrow in Australia for Europe.

If we really care about dates and times... well, then that's why you have datetime fields.

I'm sorry but I have to insist: `next_run` field must be of type `Datetime` or problems will never end. :shrug:

Reply:
@Yajo there's no TZ here. You always work with UTC and this time is in UTC and it must be processed that way. Your solution blows away the current scheme. Please restrict your review to possible problems to this approach (maybe the comparison is not done in UTC, but it should), but don't come again to change the approach.

Reply:
Yes, I'm telling possible problems. :grin: The data model is wrong and must change...

You can, however, add a new Datetime field (ex. `next_run_dt`) and provide a fallback for other addons using a computed field that just returns the date part in `next_run`.

The problem is: what is, exactly, `2020-01-01`?

- For you, it's `2020-01-01 00:00:00 UTC+02:00`.
- For somebody in Fiji it is `2020-01-01 00:00:00 UTC+12:00`.
- For somebody in Samoa, it is `2020-01-01 00:00:00 UTC-11:00`.

So, when `datetime.today()` returns `2020-01-01` for the BCE, it might return a different date elsewhere. Thus, **there is an implicit TZ** in a date value.

Then, if you convert `2020-01-01` blidly into `2020-01-01 17:00:00 UTC`, it could happen that you are referring to `17:00 UTC` for today, for tomorrow or for yesterday, because the 1st half of that datetime value (`2020-01-01`) is timzeone-aware implicitly.

To make it even more fun, Odoo stores data as UTC, but displays it localized to the user, so an Odoo instance for a multinational company could display different execution dates (even if it happened to say the same time) on different timezones, both for this model and for `ir.cron`. You might be obtaining currency rates for yesterday if you're in Samoa and run the wizard, for example.

The fact that a date and time fields alone lack timezone info makes them unreliable data types when comparing to an absolute time.

So, in summary, if we don't start using a datetime field, we'll just be switching one problem for another.

Reply:
I guess `next_run` to be datetime is proper way to do things, as well as probably making cron to be run every hour. Yet in any case Odoo itself can have only 1 rate per date, and that date is w/o timezone, which kind-a makes it complex.